codec-http/src/main/java/io/netty/handler/codec/http/cookie/ServerCookieDecoder.java

```diff
@@ -152,4 +152,23 @@ private ServerCookieDecoder(boolean strict) {
 
         return cookies;
     }
+
+    public static class ServerCookieDecoderBuilder {
+        private boolean strict;
+        private boolean reportError;
+
+        public ServerCookieDecoderBuilder strict(boolean strict) {
+            this.strict = strict;
+            return this;
+        }
+
+        public ServerCookieDecoderBuilder reportError(boolean reportError) {
+            this.reportError = reportError;
+            return this;
+        }
+
+        public ServerCookieDecoder build() {
+            return new ServerCookieDecoder(strict, reportError);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Avoid instantiation through private constructors from outside of the constructors class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AAccessorClassGeneration)